itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java

```diff
@@ -563,7 +563,21 @@ public void testEventCounts(String dbName, long fromEventId, Long toEventId, Int
   }
 
   public boolean isAcidEnabled() {
-    return hiveConf.getBoolVar(HiveConf.ConfVars.HIVE_SUPPORT_CONCURRENCY);
+    if (hiveConf.getBoolVar(HiveConf.ConfVars.HIVE_SUPPORT_CONCURRENCY) &&
+        hiveConf.getVar(HiveConf.ConfVars.HIVE_TXN_MANAGER).equals("org.apache.hadoop.hive.ql.lockmgr.DbTxnManager")) {
+      return true;
+    }
+    return false;
+  }
+
+  public void disableAcid() {
+    hiveConf.setBoolVar(HiveConf.ConfVars.HIVE_SUPPORT_CONCURRENCY, false);
+    hiveConf.setVar(HiveConf.ConfVars.HIVE_TXN_MANAGER, "org.apache.hadoop.hive.ql.lockmgr.DummyTxnManager");
+  }
+
+  public void enableAcid() {
+    hiveConf.setBoolVar(HiveConf.ConfVars.HIVE_SUPPORT_CONCURRENCY, true);
+    hiveConf.setVar(HiveConf.ConfVars.HIVE_TXN_MANAGER, "org.apache.hadoop.hive.ql.lockmgr.DbTxnManager");
```

Comment:
Is it enough to do this? I think, TxnDbUtil.prepDb should be done, otherwise, it won't work. As the modified test suit is migration case, replica warehouse instance init has initialised it and so it worked. But won't work otherwise.

Reply:
It is working for this test because, replica warehouse is acid enabled and it initialises the derby DB which is shared by both warehouse instances. But, if both primary and replica are acid disabled, then this won't work if you dynamically change this config.

Reply:
ok ..when ever its required ..it can be done ..currently i don't see any reason to add extra init if its not required

Reply:
sure. Can we add a comment about this in this method? So that later someone need not wonder why is it not working.

Reply:
removed it from warehouse instance and added to the specific test